src/engine/SCons/Node/PythonTests.py

```diff
@@ -98,18 +97,43 @@ def test_get_csig(self):
         csig = v3.get_csig(None)
         assert csig == 'None', csig
 
+
 class ValueNodeInfoTestCase(unittest.TestCase):
     def test___init__(self):
         """Test ValueNodeInfo initialization"""
         vvv = SCons.Node.Python.Value('vvv')
         ni = SCons.Node.Python.ValueNodeInfo()
 
+
 class ValueBuildInfoTestCase(unittest.TestCase):
     def test___init__(self):
         """Test ValueBuildInfo initialization"""
         vvv = SCons.Node.Python.Value('vvv')
         bi = SCons.Node.Python.ValueBuildInfo()
 
+
+class ValueMemoTestCase(unittest.TestCase):
+    def test___init__(self):
+        """Test memoization"""
+        # First confirm that ValueWithMemo does memoization.
+        value1 = SCons.Node.Python.ValueWithMemo('vvv')
+        value2 = SCons.Node.Python.ValueWithMemo('vvv')
+        assert value1 is value2
+
+        # Next confirm that ValueNodeInfo.str_to_node does memoization using
+        # the same cache as ValueWithMemo.
+        ni = SCons.Node.Python.ValueNodeInfo()
+        value3 = ni.str_to_node('vvv')
+        assert value1 is value3
+
+
+class BuiltValueNoMemoTestCase(unittest.TestCase):
```

Comment:
Pretty sure you can actually move this to test_built_value(self) in the ValueMemoTestCase